### ISSUE TITLE
Using upload url parameter on onSuccess function

### ISSUE THREAD

**Issue by dogukanoksuz** [kqito/use-tus#46]
When i try to access upload object on onSuccess function i get undefined error.

I need to get upload.url parameter to get file information after upload completed succesfully.

```
useEffect(() => {
  if (!upload) {
    return
  }

  setUrl(upload.url as string)
  console.log(upload.url)
}, [upload])
```

This sets url to null

```
onSuccess: () => {
  setUploading(false)
  console.log(upload)
  
  apiService.post("/upload_info", {
    key: upload.url
  })
}
```

This is also null.

Thanks for creating great library

**Comment by kqito** [comment]
Thank you for your reports. 

After conducting some research, I understand that React is out of sync because it does not immediately re-render after the tus-js-client internally updates such as the 'url' property. 

This behavior is not unexpected. I will address this issue soon.

**Comment by kqito** [comment]
I'm sorry for the delayed response. 

I have created a PR that should address this issue. Please check the changes in README.md to confirm whether it resolve this issue on your use case. If these changes look good to you, so I plan to release them as a new minor version.

#47

**Comment by dogukanoksuz** [comment]
Looking good. Thanks for helping out!

**Comment by kqito** [comment]
#47 has been released as v0.8.0 👏 

https://github.com/kqito/use-tus/releases/tag/v0.8.0

Please update to this latest version and confirm if the issue has been resolved. If the updated version solves your issue, kindly close this issue thread.

**Comment by dogukanoksuz** [comment]
https://github.com/limanmys/next/commit/3fb7c96ca75fc12721557eb36b051533a2b9cce6

Tried it on my project and worked well. Thanks for helping and creating great library! 

Have a great day
